### DISCUSSION TITLE
WULPUS bring-up and assembly Q&A

### DISCUSSION THREAD

**Discussion by Sergio5714** [pulp-bio/wulpus#20] [Q&A]
A test discussion for the questions related to WULPUS assembly and bring-up

**Comment by Ethan010722** [comment]
![3](https://github.com/user-attachments/assets/b1b06e19-0f7a-428e-85d5-bcc0f7221304)
![us2](https://github.com/user-attachments/assets/a82e10d6-dc0a-4586-8bc3-220e8a672832)
Hello, Author,

I am currently experiencing an issue where no signal is detected during usage, as shown in the attached image. When I start the acquisition, I tested the TR1, TR2, TR6, and TR7 test points, and there is no voltage. However, my HV15V output is normal, so I am not sure what the issue might be.

Could you please help me understand why this might be happening and how to resolve it?

Thank you for your assistance!

**Reply by Sergio5714** [reply]
Hello @Ethan010722 ,
Thank you for the question.

- It seems like your GUI is actually receiving some sampled data. This is great.
- To record any back scattered echoes, the transducer should be coupled to some media (phantom with scatters, water bath, etc). What do you use in this case?
- Could you please clarify what settings you are using (box TX/RX settings and general settings)?

There could be multiple issues. I suspect either you TX/RX config is incorrect or HV MUX switching timing needs to be tuned (if not tuned, HV MUX disconnects the pulser too early and the excitation signal cant reach the transducer).

**Reply by Ethan010722** [reply]
Dear Sergio,

Thank you very much for your prompt and detailed response. I truly appreciate your help and expertise.

As per your request, I've reviewed my configuration parameters. I've attached a screenshot of my current settings below for your reference:

From my perspective, these settings seem to be in order. However, when I look at the real-time image acquisition, it appears that there's no signal being detected. This discrepancy between the seemingly correct configuration and the lack of signal is what's puzzling me.

I'm not sure where the problem lies. It could be something I'm overlooking, or perhaps an issue with how the system is interpreting these settings. 

If you could take a look at these parameters and provide any insights or suggestions, I would be incredibly grateful. Your expertise could be crucial in helping me resolve this issue.

Best regards,
Ethan
![a](https://github.com/user-attachments/assets/a80307a9-20c3-4374-93fe-4d6b913d408c)
![b](https://github.com/user-attachments/assets/4ace8008-08ce-453b-8cca-92d0f4d7e5dd)
![c](https://github.com/user-attachments/assets/4609f801-367c-4248-88ed-48e76c1d2e38)

**Reply by Ethan010722** [reply]
Interestingly, when I detached the ultrasonic transducer, the system's acquired image didn't change at all...I don't know what is wrong with it..

**Reply by Sergio5714** [reply]
Thank you for your message and observations. Your configuration parameters seems ok to me.

> Interestingly, when I detached the ultrasonic transducer, the system's acquired image didn't change at all...I don't know what is wrong with it..

This can indicate that HV MUX switches to RX mode too early. As I described above, if it happens, the transducers will not be excited, since the HV pulses will not propagate to the transducer.

To debug this, do the following:

- Activate all channels for RX and TX (now you have only channel 7 active for RX and TX). Put a decent RX gain of e.g. 10 dB.
- Connect an oscilloscope (capable of withstanding +15-20 V pulses) to the transducer connector.
- Iteratively start acquisitions and observe the signal in the oscilloscope:
- If no excitation signal is presented, increase the parameter HV MUX RX start time [us] (in a step of 1 us). Repeat the measurements.

If the problem is with the HV MUX RX start time, at some point you will see an excitation signal. Also, in the sampled data you will see a switching artifact as it is described in section 3.4 of the User Manual.

Please, go through these steps and let me know if they solve the problem :)

P.S: I am also attaching a Fig. 4.4 from the User Guide (Rev 1) that explains the key event of the WULPUS acquisition. With the steps above, we are trying to tune event Sb that is responsible to switch HV MUX to RX mode.

![image](https://github.com/user-attachments/assets/5a249e82-9fe3-4624-a3f1-229817e0c953)

**Reply by Ethan010722** [reply]
Hello Sergio,

Thank you so much for your response! Your explanation has been very helpful.

I followed your debugging method and found some issues:

1. There is a signal in the transmission channel, but it does not show up on the graph.
2. Even if I only enabled 7 channels, there is still a signal on channels 1, 2, 3, 4, 5, and 6. I don't know the reason.
3. When I switched to ultrasound mode, no matter which channel I connected the transducer to, the image always appeared on channel 7.

Below are some pictures from the debugging process. I still don't know where the problem is. I am really grateful for your selfless help and I am very touched by it.
![88888](https://github.com/user-attachments/assets/bc1fe77d-7698-4104-a5fa-ad6bdeade36c)
![777777](https://github.com/user-attachments/assets/73dbb8b3-ce09-447a-b860-907d462cf74e)

**Reply by Sergio5714** [reply]
Hello @Ethan010722 !
Thank you for your reply.

> I followed your debugging method and found some issues:
> 
> 1. There is a signal in the transmission channel, but it does not show up on the graph.
If you measure the TX signal with the oscilloscope but do not see it the same way in received signal, it is fine.
T/R receive switch automatically blocks HV pulses and transmits only minor artifacts. Still, due to the high gain, these artifacts are amplified and cause saturation of raw data at the beginning.
> 2. Even if I only enabled 7 channels, there is still a signal on channels 1, 2, 3, 4, 5, and 6. I don't know the reason.
> 3. When I switched to ultrasound mode, no matter which channel I connected the transducer to, the image always appeared on channel 7.

This is very weird, and I can't explain it at the moment. Probably, something is wrong with the HV MUX. Please check the following points:

1. If you only tune HV MUX RX start time, does the signal change?
2. After changing the TX/RX configurations, do you run all the cells below in the Jupyter notebook?
When your GUI is already running, changing RX/TX configurations has no effect.
3. If the points above do not help, you would need to check your HV PCB assembly. It is possible that you accidentally shortened RX and TX lines. Otherwise, you would need to probe the SPI signals that control the HV MUX and Power Supplies.

**Comment by Sergio5714** [comment]
@Ethan010722, is your problem finally solved? Should we close the discussion?
